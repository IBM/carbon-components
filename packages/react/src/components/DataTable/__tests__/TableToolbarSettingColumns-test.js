/**
 * Copyright IBM Corp. 2016, 2019
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { TableToolbarSettingColumns } from '..';

describe('DataTable.TableToolbarSettingColumns', () => {
  const mockOnChange = jest.fn();
  const mockHandleMenuItemFocus = jest.fn();

  const columns = [
    { key: 'column1', header: 'Column 1' },
    { key: 'column2', header: 'Column 2' },
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render - no columns defined', () => {
    const wrapper = mount(
      <TableToolbarSettingColumns />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render - some columns selected', () => {
    const initialSelected = ['column2'];
    const wrapper = mount(
      <TableToolbarSettingColumns
        columns={columns}
        initialSelected={initialSelected}
        onChange={mockOnChange}
        handleMenuItemFocus={mockHandleMenuItemFocus}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle onChange', () => {
    const initialSelected = ['column1'];
    const wrapper = mount(
      <TableToolbarSettingColumns
        columns={columns}
        initialSelected={initialSelected}
        onChange={mockOnChange}
        handleMenuItemFocus={mockHandleMenuItemFocus}
      />
    );

    // uncheck column1
    wrapper.find('input[id="column1"]').simulate('change', { target: { checked: false } });
    expect(mockOnChange.mock.calls.length).toEqual(1);
    expect(mockOnChange.mock.calls[0]).toEqual([[]]);

    jest.clearAllMocks();

    // check column2
    wrapper.find('input[id="column2"]').simulate('change', { target: { checked: true } });
    expect(mockOnChange.mock.calls.length).toEqual(1);
    expect(mockOnChange.mock.calls[0]).toEqual([['column2']]);
  });
});
