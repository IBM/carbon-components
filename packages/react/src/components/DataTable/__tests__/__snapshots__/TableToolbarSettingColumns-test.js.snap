// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableToolbarSettingColumns should render - no columns defined 1`] = `
<ForwardRef
  columns={Array []}
  selectedColumns={Array []}
  translateWithId={[Function]}
>
  <ForwardRef
    title="Edit column"
  >
    <li
      className="bx--table-toolbar-menu__title"
      role="menuitem"
    >
      Edit column
    </li>
  </ForwardRef>
</ForwardRef>
`;

exports[`DataTable.TableToolbarSettingColumns should render - some columns selected 1`] = `
<ForwardRef
  columns={
    Array [
      Object {
        "header": "Column 1",
        "key": "column1",
      },
      Object {
        "header": "Column 2",
        "key": "column2",
      },
    ]
  }
  handleMenuItemFocus={[MockFunction]}
  initialSelected={
    Array [
      "column2",
    ]
  }
  onChange={[MockFunction]}
  selectedColumns={Array []}
  translateWithId={[Function]}
>
  <ForwardRef
    title="Edit column"
  >
    <li
      className="bx--table-toolbar-menu__title"
      role="menuitem"
    >
      Edit column
    </li>
  </ForwardRef>
  <ForwardRef
    key="column1"
  >
    <li
      className="bx--table-toolbar-menu__option"
      role="menuitem"
    >
      <ForwardRef(Checkbox)
        data-table-toolbar-focusable={true}
        defaultChecked={false}
        id="column1"
        indeterminate={false}
        labelText="Column 1"
        onChange={[Function]}
        onKeyDown={[MockFunction]}
      >
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            className="bx--checkbox"
            data-table-toolbar-focusable={true}
            defaultChecked={false}
            id="column1"
            onChange={[Function]}
            onKeyDown={[MockFunction]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label"
            htmlFor="column1"
            title={null}
          >
            <span
              className="bx--checkbox-label-text"
            >
              Column 1
            </span>
          </label>
        </div>
      </ForwardRef(Checkbox)>
    </li>
  </ForwardRef>
  <ForwardRef
    key="column2"
  >
    <li
      className="bx--table-toolbar-menu__option"
      role="menuitem"
    >
      <ForwardRef(Checkbox)
        data-table-toolbar-focusable={true}
        defaultChecked={false}
        id="column2"
        indeterminate={false}
        labelText="Column 2"
        onChange={[Function]}
        onKeyDown={[MockFunction]}
      >
        <div
          className="bx--form-item bx--checkbox-wrapper"
        >
          <input
            className="bx--checkbox"
            data-table-toolbar-focusable={true}
            defaultChecked={false}
            id="column2"
            onChange={[Function]}
            onKeyDown={[MockFunction]}
            type="checkbox"
          />
          <label
            className="bx--checkbox-label"
            htmlFor="column2"
            title={null}
          >
            <span
              className="bx--checkbox-label-text"
            >
              Column 2
            </span>
          </label>
        </div>
      </ForwardRef(Checkbox)>
    </li>
  </ForwardRef>
</ForwardRef>
`;
